### ISSUE TITLE
Add the ability to deal with Optional fields

### ISSUE THREAD

**Issue by PhilippMDoerner** [enthus1ast/nisane#3]
I recently stumbled over this.
When you do e.g. a LEFT JOIN, chances are there will be rows where there a column with a NULL value.
Naturally within the "Row" object you get in your nim code that thing will just be an empty string, so normal conversion is still possible.

However, it would be more appropriate to have an Optional there instead of a field that'll get an empty string. 
So for example, if I have such a query:

```sql
  SELECT 
    char.name AS charName, 
    loc.name AS locationName
  FROM character char
  LEFT JOIN location loc ON char.current_location_id = loc.id
```

Where a character may have been assigned to a specific location or to none, I would want to be able to use such a type:
```nim
type QueryObj = object
  charName: string
  locationName: Option[string]
```

As a first idea, I think converting any empty string `""` straight into "none(<WhateverTypeTheFieldHas>)" would be a valid approach for this. 

Does this feature make sense?

On an unrelated note, I have seen that nisane has the ability to deal with custom types as long as a `to<X>` proc is provided.
Could that be used for types with generic fields (such as optionals for example) ?

**Comment by enthus1ast** [comment]
yes you can define a toXX proc that handle these specific cases.
But i'll also look into this (after correctly fixing #2)
